dotlottie-rs/src/state_machine/parser/mod.rs

```diff
@@ -0,0 +1,223 @@
+}
+
+#[derive(Deserialize, Debug, PartialEq)]
+pub struct BooleanEventJson {
+    pub value: bool,
+}
+
+#[derive(Deserialize, Debug, PartialEq)]
+pub struct OnCompleteEventJson {}
+
+#[derive(Deserialize, Debug, PartialEq)]
+pub struct OnPointerDownEventJson {
+    pub target: Option<String>,
+}
+
+#[derive(Deserialize, Debug, PartialEq)]
+pub struct OnPointerUpEventJson {
+    pub target: Option<String>,
+}
+
+#[derive(Deserialize, Debug, PartialEq)]
+pub struct OnPointerEnterEventJson {
+    pub target: Option<String>,
+}
+
+#[derive(Deserialize, Debug, PartialEq)]
+pub struct OnPointerExitEventJson {
+    pub target: Option<String>,
+}
+
+#[derive(Deserialize, Debug, PartialEq)]
+pub struct OnPointerMoveEventJson {
+    pub target: Option<String>,
+}
+
+#[derive(Deserialize, Debug, PartialEq)]
+pub struct TransitionJson {
+    pub r#type: TransitionJsonType,
+    pub from_state: u32,
+    pub to_state: u32,
+    pub guards: Option<Vec<TransitionGuardJson>>,
+    pub numeric_event: Option<NumericEventJson>,
+    pub string_event: Option<StringEventJson>,
+    pub boolean_event: Option<BooleanEventJson>,
+    pub on_complete_event: Option<OnCompleteEventJson>,
+    pub on_pointer_down_event: Option<OnPointerDownEventJson>,
+    pub on_pointer_up_event: Option<OnPointerUpEventJson>,
+    pub on_pointer_enter_event: Option<OnPointerEnterEventJson>,
+    pub on_pointer_exit_event: Option<OnPointerExitEventJson>,
+    pub on_pointer_move_event: Option<OnPointerMoveEventJson>,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct ListenerJson {
+    pub r#type: ListenerJsonType,
+    pub target: Option<String>,
+    pub action: Option<String>,
+    pub value: Option<StringNumberBool>,
+    pub context_key: Option<String>,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct ContextJson {
+    pub r#type: ContextJsonType,
+    pub key: String,
+    pub value: StringNumberBool,
+}
+
+impl ContextJson {
+    pub fn new() -> ContextJson {
+        Self {
+            r#type: ContextJsonType::String,
+            key: String::new(),
+            value: StringNumberBool::String(String::new()),
+        }
+    }
+}
+
+impl Default for ContextJson {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+#[derive(Deserialize, Debug)]
+pub struct StateMachineJson {
+    pub descriptor: DescriptorJson,
+    pub states: Vec<StateJson>,
+    pub transitions: Vec<TransitionJson>,
+    pub listeners: Vec<ListenerJson>,
+    pub context_variables: Vec<ContextJson>,
+}
+
+pub fn state_machine_parse(json: &str) -> Result<StateMachineJson, StateMachineError> {
+    let result: Result<StateMachineJson, serde_json::Error> = serde_json::from_str(json);
+
+    match result {
+        Ok(state_machine_json) => Ok(state_machine_json),
+        Err(err) => Err(StateMachineError::ParsingError {
+            reason: format!(
+                "Error parsing state machine definition file: {:?}",
+                err.to_string()
+            ),
+        }),
+    }
```

Comment:
the Result can be mapped instead of wrapped. 

```suggestion
    serde_json::from_str(json).map_err(|err| StateMachineError::ParsingError {
        reason: format!("Error parsing state machine definition file: {}", err)
    })
```